### ISSUE TITLE
AXFR from tinydnsbackend does not respect location specifiers

### ISSUE THREAD

**Issue by rcalpha** [PowerDNS/pdns#12083]
Records that have client location is not being listed in an AXFR transfer.

However the missing records is resolved successfully using a query, with the expected result both cases, inside and outside client location. The issue is in AXFR only not the query.

To reproduce the issue, I used DJBDNS to make an AXFR for control test. I compared the AXFR using PowerDNS with tinyDNS backend, and PowerDNS has missing entries. Both using the same data.cdb file with entry that use client location field.

--------------------
PowerDNS: 4.5.3-1
backend: tinyDNS
DNSSEC: no
OS: Ubuntu server 22.04.1

**Comment by cyclops1982** [comment]
@rcalpha Would you mind sharing how you did the AXFR from tinydns? 
`tinydns` itself does not support it, only the `axfrdns` program.

**Comment by Habbie** [comment]
The [docs](https://doc.powerdns.com/authoritative/backends/tinydns.html#tinydns-locations) for `tinydns-locations` currently say:

> Enable or Disable location support in the backend. Changing the value to ‘no’ will make the backend ignore the locations. This then returns all records. When the setting is changed to ‘no’ an AXFR will also return all the records. With the setting on ‘yes’ an AXFR will only return records without a location.

So, it looks like the current behaviour is correct in the sense that it matches the documentation.

Because of the way things are designed inside PowerDNS, making AXFR actually location aware is not easy right now.

I'm open to changing the behaviour so that `tinydns-locations=yes` also returns all records, or perhaps adding a third value to this option that would allow the AXFR behaviour to be specified independently of normal query behaviour.

**Comment by rcalpha** [comment]
Sorry the delay, quite busy,

About the questions:

> Does the AXFR become correct if you set tinydns-locations=no
No, it list all records ignoring the location of the client

> Because of the way things are designed inside PowerDNS, making AXFR actually location aware is not easy right now.
I understand

I use axfr to validate results of DNS entries with location, to compare from inside and outside view.
To do test, I use axfrdns from djbdns/dbndns

I'm currently finding a migration path away from dbndns/djbdns because there is an issue in ubuntu 22.04 which includes again the packages. Even compiling by my self for ubuntu 22.04 have the same issue. In ubuntu 20.04 the recompiled by me works. dbndns/djbdns is not included in ubuntu 18.04 and 20.04

**Comment by Habbie** [comment]
Right. I guess you'll have to dig individual records to test, instead.
